src/redux/reducers/planReducer.js

```diff
@@ -0,0 +1,123 @@
+    case PlanActions.GET_PLANS_FAILURE:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.GET_PLANS_SUCCESS:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced,
+        plans: action.plans
+      };
+    case PlanActions.ADD_PLAN_REQUEST:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.ADD_PLAN_FAILURE:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.ADD_PLAN_SUCCESS:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.UPDATE_PLAN_REQUEST:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.UPDATE_PLAN_FAILURE:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.UPDATE_PLAN_SUCCESS:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.DELETE_PLAN_REQUEST:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.DELETE_PLAN_FAILURE:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.DELETE_PLAN_SUCCESS:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced,
+        plans: action.plans
+      };
+    case PlanActions.GET_PLAN_INFO_REQUEST:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.GET_PLAN_INFO_FAILURE:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced
+      };
+    case PlanActions.GET_PLAN_INFO_SUCCESS:
+      return {
+        ...state,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
+        isSynced: action.isSynced,
+        plans: action.plans
+      };
+    default:
+      return state;
+  }
```

Comment:
I think more of the logic regarding editing the state needs to go here (take it out of all of the actions, which shouldn't be directly manipulating the store's state). This applies for all of the newly added reducers.

Reply:
See my comment above.